src/main/java/com/baidu/hugegraph/driver/HugeClientBuilder.java

```diff
@@ -0,0 +1,191 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.driver;
+
+public class HugeClientBuilder {
+
+    private static final int CPUS = Runtime.getRuntime().availableProcessors();
+    private static final String DEFAULT_HTTP_PREFIX = "http";
+    private static final int DEFAULT_IDLE_TIME = 30;
+    private static final int DEFAULT_MAX_CONNS = 4 * CPUS;
+    private static final int DEFAULT_MAX_CONNS_PER_ROUTE = 2 * CPUS;
+    private static final int DEFAULT_TIMEOUT = 20;
+
+    private String graph;
+    private int idleTime;
+    private int maxConns;
+    private int maxConnsPerRoute;
+    private String password;
+    private String protocol;
+    private int timeout;
+    private String trustStoreFile;
+    private String trustStorePassword;
+    private String url;
+    private String username;
+
+    public HugeClientBuilder() {
+        this.username = "";
+        this.password = "";
+        this.timeout = DEFAULT_TIMEOUT;
+        this.maxConns = DEFAULT_MAX_CONNS;
+        this.maxConnsPerRoute = DEFAULT_MAX_CONNS_PER_ROUTE;
+        this.protocol = DEFAULT_HTTP_PREFIX;
+        this.trustStoreFile = "";
+        this.trustStorePassword = "";
+        this.idleTime = DEFAULT_IDLE_TIME;
+    }
+
+    public HugeClientBuilder(String url, String graph) {
+        if (url == null || url.isEmpty()) {
+            throw new IllegalArgumentException(String.format("Expect a string value" +
+                      " as the url parameter argument, but got: %s", url));
+        }
+        if (graph == null || graph.isEmpty()) {
+            throw new IllegalArgumentException(String.format("Expect a string value" +
+                    " as the graph name parameter argument, but got: %s", graph));
+        }
+        this.url = url;
+        this.graph = graph;
+        this.username = "";
+        this.password = "";
+        this.timeout = DEFAULT_TIMEOUT;
+        this.maxConns = DEFAULT_MAX_CONNS;
+        this.maxConnsPerRoute = DEFAULT_MAX_CONNS_PER_ROUTE;
+        this.protocol = DEFAULT_HTTP_PREFIX;
+        this.trustStoreFile = "";
+        this.trustStorePassword = "";
+        this.idleTime = DEFAULT_IDLE_TIME;
+    }
+
+    public HugeClient build() {
+        if (this.url == null) {
+            throw new IllegalArgumentException("The url parameter is invalid and" +
+                                               " cannot be null");
+        }
+        if (this.graph == null) {
+            throw new IllegalArgumentException("The graph name parameter is invalid" +
+                                               " and cannot be null");
+        }
+        return new HugeClient().create(this);
+    }
+
+    public HugeClientBuilder configGraph(String graph) {
```

Comment:
just graph(String graph) is enough，it's our style

Reply:
![image](https://user-images.githubusercontent.com/19245404/85558066-fa3f8480-b65a-11ea-90bf-382552fc333f.png)

by design